serverlessworkflow/modules/ROOT/pages/core/custom-functions-support.adoc

```diff
@@ -205,6 +205,112 @@ public class MyInterfaceOrClass {
 ====
 
+== Custom function types
+
+You can add your our custom types by using Kogito add-on mechanism. As predefined custom types like `sysout` or `java`, the custom type identifier is the prefix of the operation field of the function definition. 
+
+Kogito add-ons relies on link:https://es.quarkus.io/guides/writing-extensions[Quarkus extensions] mechanism. And add-on consist of at least two Maven projects:
+- deployment, which is responsible for generating code required for the extension to work.
+- runtime, which include the non generated classes that are required for the extension to work. 
+
+In the case of a serverless workflow custom type:
+
+- Deployment project is expected to configure the work item handler that will be used during runtime to perform the logic associated to the custom type. It must contain a Java class that inherits from `WorkItemTypeHandler`. Its responsibilities are to indicate the custom type identifier (the operation prefix, as indicated earlier) and to setup the `WorkItemNodeFactory` instance passed as parameter of `fillWorkItemHandler` method. That instance is included in the Kogito process definition for that Workflow. As part of this setup, you must indicate the name of the  `WorkItemNodeFactory`. You might also provide any relevant metadata for that handler, if needed. 
+- Runtime project consists of a `WorkflowWorkItemHandler` implementation, which name must match the one provided to `WorkItemNodeFactory` during the deployment phase, and a `WorkItemHandlerConfig` bean that register that handler with that name. 
+ When a Serverless workflow function is called, Kogito identifies the proper `WorkflowWorkItemHandler` instance to be used for that function type (using the handler name associated to that type by the deployment project) and then invokes `internalExecute` method. The `Map` parameter contains the function arguments defined in the Workflow and the `WorkItem` parameter contains the  metadata information added to the handler by the deployment project. Hence, `executeWorkItem` implementation has access to all the information needed to perform the computational logic intended for that custom type. 
+
+
+=== Custom function type example. 
+
+Lets assume we want to interact, from a workflow file, with a legacy RPC server as the one defined link:https://github.com/kiegroup/kogito-examples/tree/stable/serverless-workflow-examples/serverless-workflow-custom-type/serverless-workflow-custom-rpc-server[this project]. This legacy server support four simple arithmetic operations: add, minus, multiply an divide, that can be invoked using a custom RPC protocol. Since this a non common protocol, workflow cannot handle them by using any of the predefined Serveless workflow function types. The available options are to use a Java Service, which invokes a Java class that knows how to interact with the server; or define a custom type that knows how to interact with the service. 
```

Comment:
```suggestion
Let's assume we want to interact, from a workflow file, with a legacy RPC server as the one defined link:https://github.com/kiegroup/kogito-examples/tree/stable/serverless-workflow-examples/serverless-workflow-custom-type/serverless-workflow-custom-rpc-server[this project]. This legacy server supports four simple arithmetic operations: add, minus, multiply and divide, that can be invoked using a custom RPC protocol. Since this a non common protocol, workflow cannot handle them by using any of the predefined Serverless workflow function types. The available options are to use a Java Service, which invokes a Java class that knows how to interact with the server; or define a custom type that knows how to interact with the service. 
```